### PR TITLE
LogReporting 2

### PR DESCRIPTION
**Pull request by cdillard-NewRelic**
New Log Reporting Feature Branch includes some new tickets and will be used as the accumulation branch for the rest of the required changes for LogReporting / config.

Includes: 
https://new-relic.atlassian.net/browse/NR-170090  : [iOS] Remote Logging should cache log request data to disk
https://new-relic.atlassian.net/browse/NR-227297 : [iOS] [Remote Logging] Support logError
https://new-relic.atlassian.net/browse/NR-227300 : [iOS] [Remote Logging] Support logAttributes
https://new-relic.atlassian.net/browse/NR-266266 : [iOS] Implement new endpoint header type now included in connect response
https://new-relic.atlassian.net/browse/NR-265385: [iOS] Implement sampling strategy
https://new-relic.atlassian.net/browse/NR-272767:  [iOS] Refactor existing Agent Logs to be disabled for this release

### CODE REVIEW

**Review by mbruin-NR — APPROVED** [review]

**Review by mbruin-NR — DISMISSED** [review]

**Comment by cdillard-NewRelic** [comment]
@smalsam-newr @mbruin-NR Please approve.

**Review by mbruin-NR — APPROVED** [review]

File: Agent/Harvester/NRMAHarvesterConnection.h [hunk 1]
```diff
@@ -13,13 +13,16 @@
 #import "NRMAConnection.h"
 #import "NRMAOfflineStorage.h"
 
+#define kCOLLECTOR_CONNECT_URI         @"/mobile/v5/connect"
+#define kCOLLECTOR_DATA_URL            @"/mobile/v3/data"
```
Comment (smalsam-newr) [inline Agent/Harvester/NRMAHarvesterConnection.h:17]: These are also defined in Constants.h. Is it possible to consolidate the two uses of these down to one, either here or there?

File: Agent/Public/NRLogger.h [hunk 2]
```diff
@@ -132,19 +147,11 @@ withMessage:(NSString *)message;
  */
 + (void)setLogTargets:(unsigned int)targets;
 
-/*
-Configure the New Relic logging API.
+// For internal use only. Do not use.
++ (void)setLogIngestKey:(NSString*)key;
```
Comment (smalsam-newr) [inline Agent/Public/NRLogger.h:151]: Consider using an internal header here, and adding a category, so we don't have to directly expose our internal APIs.

File: Test Harness/NRTestApp/NRAPI-Info.plist [hunk 3]
```diff
@@ -6,8 +6,6 @@
 	<string></string>
 	<key>collectorAddress</key>
 	<string></string>
-	<key>logEntityGuid</key>
```
Comment (smalsam-newr) [inline Test Harness/NRTestApp/NRAPI-Info.plist:9]: Is this no longer something that can be provided at runtime, like the API Key?

**Review by smalsam-newr — APPROVED** [review]

**Pull request merged**